[CLI] Add a helper page for movefmt (#617)
aptos-labs/developer-docs: apps/nextra/pages/en/build/cli/formatting-move-contracts.mdx

# Formatting Move Contracts

`movefmt` is a formatter tool that makes Move code much easier to write, read, and maintain — greatly improving the development experience on Aptos.

## Installation

`movefmt` is integrated into the Aptos CLI. To begin using it, first install it using the CLI update command.

```bash filename="Terminal"
# Install movefmt for first time usage
aptos update movefmt
```

> When a newer version of `movefmt` is available, you can specify the version to be installed:

Reviewer: ```suggestion
To install a specific version of `movefmt`:
```

```bash filename="Terminal"
# Install movefmt with the target <VERSION>

> aptos update movefmt --target-version <VERSION>

Author: Unfortunately, the current updater does not support updating to the latest version with out input. Instead, it will install the version hard coded

> aptos update movefmt --target-version <VERSION>

Reviewer: Will we be updating the hard coded version whenever a new version is available?

If not, when will the hardcoded version be updated?

> aptos update movefmt --target-version <VERSION>

Author: We should do it once the formatter is more stable but since it is still beta and the update will be frequent, it is harder to catch up with it. I think we can do it before each CLI release.

> aptos update movefmt --target-version <VERSION>

Reviewer: Awesome, makes sense.

```

## Format your code

Similar to compilation and testing, you can use the following command to format the Move package:

```bash filename="Terminal"
# Format the Move package
aptos move fmt
```

Different ways of emitting the formatting result is supported:

```bash filename="Terminal"

> # Format and overwrite the move file `target.move`

Reviewer: ```suggestion
# Format and overwrite all the target move files in the package.
```

# This is the default behavior if `--emit-mode` is not explicitly specified
aptos move fmt --emit-mode=overwrite

# Print the formatting result to terminal
aptos move fmt --emit-mode=std-out

> # Print the formatting result to new files

Reviewer: Where will the new files be saved?

aptos move fmt --emit-mode=new-file

# Print the difference between before and after formatting
aptos move fmt --emit-mode=diff
```

`movefmt` also provides different options to configure how the code will be formatted.
Here is the default configuration,

> which says 1) each line can have at most 90 characters, 2) the ident is 4 spaces,

Reviewer: ```suggestion
which says 1) each line can have at most 90 characters, 2) the indent is 4 spaces,
```

> which says 1) each line can have at most 90 characters, 2) the ident is 4 spaces,

Reviewer: This description might look better as comments right next to the config values, rather than having to manually count and match.

> 3) each tab is identical 4 spaces and 4) when a tab is inserted, it will be automatically replaced by 4 spaces.

Reviewer: ```suggestion
3) each tab is identical to 4 spaces and 4) when a tab is inserted, it will be automatically replaced by 4 spaces.
```

```

> max_width = 90 // each line can have at most 90 characters

Reviewer: ```suggestion
max_width = 90  # each line can have at most 90 characters
```

Reviewer: Comments in toml begin with `#`. Same for the remaining lines here.
